packages/builders/src/components/ActionRow.ts

```diff
@@ -12,14 +12,18 @@ export type ActionRowComponent = ButtonComponent | SelectMenuComponent;
 /**
  * Represents an action row component
  */
-export class ActionRow<T extends ActionRowComponent = ActionRowComponent> implements Component {
+export class ActionRow<T extends ActionRowComponent = ActionRowComponent> extends Component<APIActionRowComponent> {
 	public readonly components: T[] = [];
-	public readonly type = ComponentType.ActionRow;
 
 	public constructor(data?: APIActionRowComponent & { type?: ComponentType.ActionRow }) {
+		super({ type: ComponentType.ActionRow, ...data });
 		this.components = (data?.components.map(createComponent) ?? []) as T[];
```

Comment:
Do we want to have `ar.data.components` and `ar.components`, even mismatching in values after using `ar.setComponents()` or `ar.addComponents()`?

The way to fix this is by changing the `data` parameter to `{ components, ...data }`.

Also make sure to edit the extends so it omits `components` just like `UnsafeSelectMenuComponent` does.